cdap-gateway/src/main/java/co/cask/cdap/gateway/router/handlers/WrappedDiscoverable.java

```diff
@@ -40,6 +40,10 @@ public InetSocketAddress getSocketAddress() {
     return discoverable.getSocketAddress();
   }
 
+  public byte[] getPayload() {
```

Comment:
I have created a JIRA for cleaning up WrappedDiscoverable. https://issues.cask.co/browse/CDAP-7633

For this PR, I have added payload to hashCode and equals.